runtime/j9vm/javanextvmi.cpp

```diff
@@ -474,36 +519,35 @@ JNIEXPORT void JNICALL
 JVM_VirtualThreadHideFrames(JNIEnv *env, jobject vthread, jboolean hide)
 {
 	J9VMThread *currentThread = (J9VMThread *)env;
-	Assert_SC_true(IS_JAVA_LANG_VIRTUALTHREAD(currentThread, currentThread->threadObject));
 
+	Assert_SC_true(IS_JAVA_LANG_VIRTUALTHREAD(currentThread, currentThread->threadObject));
 	if (hide) {
 		Assert_SC_true(J9_ARE_NO_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES));
-		currentThread->privateFlags |= J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
 	} else {
 		Assert_SC_true(J9_ARE_ALL_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES));
-		currentThread->privateFlags &= ~(UDATA)J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
 	}
+	VM_VMHelpers::virtualThreadHideFrames(currentThread, hide);
 }
 #endif /* JAVA_SPEC_VERSION >= 20 */
 
 #if JAVA_SPEC_VERSION >= 21
 JNIEXPORT void JNICALL
-JVM_VirtualThreadMount(JNIEnv* env, jobject vthread, jboolean hide, jboolean firstMount)
+JVM_VirtualThreadMount(JNIEnv* env, jobject vthread, jboolean hide)
 {
 	if (hide) {
-		JVM_VirtualThreadMountBegin(env, vthread, firstMount);
+		virtualThreadMountBegin(env, vthread);
 	} else {
-		JVM_VirtualThreadMountEnd(env, vthread, firstMount);
+		virtualThreadMountEnd(env, vthread);
```

Comment:
I think names using the suffixes "Begin" and "End" just serve to muddy the waters here; "hide" and "begin vs. end" would  (just based on the names) seem unrelated.

Reply:
How should the code be changed for this feedback? begin and end represent the start and end of the virtual thread mount and unmount operations. hide toggles the switch to hide frames and disable certain JVMTI events between the begin and end operations.